src/Data/YAML/Event/Writer.hs

```diff
@@ -208,23 +208,21 @@ putNode = \docMarker -> go (-1 :: Int) (not docMarker) BlockIn
 
       DoubleQuoted    -> pfx $ T.B.singleton '"'  <> T.B.fromText (escapeDQ t) <> T.B.singleton '"'  <> contEol
 
-      -- block-style
```

Comment:
We still need to either a) throw ugly exceptions or b) silently/automatically fixup things

i.e. we should honor the chomping indicators & indentation offsets when emitting YAML -- but only when they result in valid YAML 1.2 that faithfully encodes the payload 

For instance, in particular if "Clip" was requested, but the scalar string in the constructor contains trailing empty lines, then we shall switch to `Keep`; or if `IndentAuto` was requested but the payload has leading whitespace, we shall instead emit an explicit indent-offset.

IOW, we shall take the comping/indentation indicators merely as "hints" but not as imperatives.